armory/scenarios/poisoning_sleeper_agent.py

```diff
@@ -161,4 +167,19 @@ def poison_dataset(self):
                 np.array([]),
             )
 
+        # poison_index returned by attack is the index within the target class, not the whole dataset.
+        # In addition, it may contain images that aren't actually perturbed.
+        # Find the true poison indices
+        true_poison_inds = np.array(
+            [
+                i
+                for i in range(len(self.x_clean))
+                if (self.x_clean[i] != self.x_poison[i]).all()
+            ]
+        )
+        n_target = (self.y_clean == self.target_class).sum()
+        log.info(
+            f"Actual amount of poison returned by attack: {len(true_poison_inds)} samples or {len(true_poison_inds)/n_target} percent"
+        )
+        self.poison_index = true_poison_inds
```

Comment:
earlier in the function should we be setting `self.poison_index` if it's overriden immediately with `true_poison_inds`?

Reply:
it is not necessary to set self.poison_index earlier.  I can update it.